### ISSUE TITLE
Feedback from Simon at PlainJS.com

### ISSUE THREAD

**Issue by anthonyjb** [GetmeUK/ContentTools#63]
The following items for enhancement all come from a discussion with Simon Steinberger at PlainJS.com about his experience of the editor. I think each is valid and so will be looking to make changes in line with that feedback.

I would appreciate thoughts on how to best resolve 2 and 4.

1. Links can be removed by setting the link to empty, so if you select a section of link that you no longer want to be a link then click the link tool, remove the `href` value and hit return the link will be removed.
    - Based on you're feedback I think it makes a lot of sense to allow someone to edit a link if they have the cursor within a linked section of text, in that scenario selecting the link tool should automatically select the entire link.
    - I will also look at reinstating a remove link tool, I took this out as I never used it but I can see others might expect it.
    - I think it's still valid to allow a link to removed from only part of a selection.
2. As you say if there are 2 or more links in a selection of text then editing the link simply brings up the first `href` value and lets you set the link for the entire selection.
    - I'm not sure what the best behavior here is but I can see how this might be confusing for a user, I'll have a think on this.
3.  Toggling headings and lists back to paragraphs.
    - That makes perfect sense, I'll make that change. As you say at the moment you have to use the paragraph tool.
4. Removing lists and tables. You can remove a list item or table row using the remove tool, if you want to remove the entire list or table then hold down the Ctrl (or Command key on Apple) when selecting the remove tool.
    - I think this feature probably needs to be more obvious to users, I'll have a think about how I can do that.

**Comment by bfintal** [comment]
2. I think this is the correct behavior except that the link shown shouldn't be the first `href` but maybe a label that says _multiple links_. Then overwriting that label would set the link for the entire selection.

4. Personally I prefer deleting the rows/list items one by one then when the last one gets deleted the table/list gets deleted entirely. I don't think holding ctrl would be intuitive to a large part of users.

**Comment by anthonyjb** [comment]
@bfintal Ctrl + remove support is already in place, I suppose the concern is as you say people might not ever realise it or find it intuitive.

**Comment by bfintal** [comment]
@anthonyjb I didn't know that even existed!

**Comment by bfintal** [comment]
**No. 3** The heading/subheading/paragraph buttons should look depressed if they're applied.

**Comment by anthonyjb** [comment]
@bfintal agreed, especially if they'll toggle the element back to paragraph.

**Comment by anthonyjb** [comment]
OK so I finally got around to implementing some of the changes here. This issue has been up for a while and having had time to think over the various features suggested at this time I decided just to implement the following:

- Support for link editing when the caret is within an existing link. Now if you click within a link then click the link tool the whole link will be selected for you.
- Support for headings, paragraphs and pre text being being set to applied when the an element with the same tag name is selected, and support for toggling between headings/pre-text and paragraphs. E.g press the heading tool while a heading is selected and it will revert to a paragraph.

I'm not currently looking to implement any of the other suggestions as they've not been raised since (that I'm aware of), e.g a link remove tool, a way to handle selecting multiple links.
